### PR TITLE
feat(agent): Add basic support for EKS Pods on Fargate

### PR DESCRIPTION
**Pull request by Ferenc-**

### CODE REVIEW

File: instana/agent/aws_eks_fargate.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        response = None
+        try:
+            if self.report_headers is None:
+                # Prepare request headers
+                self.report_headers = dict()
+                self.report_headers["Content-Type"] = "application/json"
+                self.report_headers["X-Instana-Host"] = self.podname
+                self.report_headers["X-Instana-Key"] = self.options.agent_key
+
+            self.report_headers["X-Instana-Time"] = str(round(time.time() * 1000))
```
[the remark this answers is not in the document]
Comment (Ferenc-) [reply instana/agent/aws_eks_fargate.py:70]: Done. Also for the rest of the agents in https://github.com/instana/python-sensor/pull/473

File: instana/singletons.py [hunk 2]
```diff
@@ -17,9 +19,12 @@
 
 # Detect the environment where we are running ahead of time
 aws_env = os.environ.get("AWS_EXECUTION_ENV", "")
+uname_output = subprocess.run(['uname -r'], shell=True, stdout=subprocess.PIPE).stdout
+aws_kernel = re.match('.+[.]amzn[0-9]+[.].+', uname_output.decode('ascii') if uname_output else '')
```
[the remark this answers is not in the document]
Comment (Ferenc-) [reply instana/singletons.py]: Removed

**Review by pvital — APPROVED** [review]
It looks good to me.  

My only suggestion is to change the c5666f3 commit message from `test: Cover EKSFargateAgent` to `feat (test): Add EKSFargateAgent test cases` since we are adding this whole set of test cases.

File: instana/collector/aws_eks_fargate.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+            with_snapshot = self.should_send_snapshot_data()
+
+            plugins = []
+            for helper in self.helpers:
+                plugins.extend(helper.collect_metrics(with_snapshot=with_snapshot))
+
+            payload["metrics"]["plugins"] = plugins
+
+            if with_snapshot is True:
```
Comment (GSVarsha) [inline instana/collector/aws_eks_fargate.py:48]: Shouldn't we just say `if with_snapshot:`  since `should_send_snapshot_data()` always returns a bool?

Comment (Ferenc-) [reply instana/collector/aws_eks_fargate.py:48]: We should. Good catch!

**Comment by Ferenc-** [comment]
> feat (test): Add EKSFargateAgent test cases

Done

**Review by GSVarsha — APPROVED** [review]
Looks perfect to me!

**Pull request merged**